rest_framework_nested/viewsets.py

```diff
@@ -29,7 +39,7 @@ def _get_parent_lookup_kwargs(self) -> dict:
         parent_lookup_kwargs = getattr(self, 'parent_lookup_kwargs', None)
 
         if not parent_lookup_kwargs:
-            serializer_class = self.get_serializer_class()
+            serializer_class = cast(GenericAPIView, self).get_serializer_class()
```

Comment:
This one I am still trying to digest. Seems as a source of confusion to the end developer.

Instead, does typing the `serializer_class` directly achieves the same result?
```suggestion
            serializer_class: Serializer = self.get_serializer_class()
```

Reply:
This is a general issue with mixin classes.

This code here in `NestedViewSetMixin` code assumes that whatever code using it always combines it with something that derives from `GenericAPIView`. But it's not actually part of the class hierarchy.

Adding the type hint to `serializer_class` doesn't solve the error from mypy, because the type error is about being unable to resolve `get_serializer_class()` method on `self`.